### PR TITLE
Adding files for Assignment-1

### PR DESCRIPTION
**Pull request by khushikdu**
Adding files for assignment-1

### CODE REVIEW

**Comment by khushikdu** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 3 |
| code_smells | 40 |
| reliability_rating | D (at least 1 Critical Bug)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

File: backend/assignments/1/src/main/java/org/services/Main.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+     */
+    public static void executeTransactions(JsonNode jsonTransactions, CountDownLatch latch) {
+        try {
+             CryptoApplication cryptoService=new CryptoApplication();
+             coins = cryptoService.loadCoinsFromCSV("src/main/resources/coins.csv");
+             traders=cryptoService.loadTraderFromCSV("src/main/resources/traders.csv");
+             transactions = cryptoService.loadTransactionsFromJSON("src/main/resources/small_transaction.json");
+             ExecutorService executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
+             for (Transaction transaction : transactions)
+                executorService.submit(new ExecuteTransaction(transaction,coins,traders, latch));
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/services/Main.java:48]: Wrap with {}

File: backend/assignments/1/src/main/java/org/services/Main.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+     *
+     * @param jsonTransactions
+     * @param latch
+     */
+    public static void executeTransactions(JsonNode jsonTransactions, CountDownLatch latch) {
+        try {
+             CryptoApplication cryptoService=new CryptoApplication();
+             coins = cryptoService.loadCoinsFromCSV("src/main/resources/coins.csv");
+             traders=cryptoService.loadTraderFromCSV("src/main/resources/traders.csv");
+             transactions = cryptoService.loadTransactionsFromJSON("src/main/resources/small_transaction.json");
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/services/Main.java:45]: Define `constants` and then use paths

File: backend/assignments/1/src/main/java/org/services/CryptoService.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    public void processAddVolumeTransaction(Transaction transaction){
+        synchronized (lock) {
+            Coin coin = getCoinBySymbol(transaction.getCoinSymbol());
+            if (coin != null) {
+                if(coin.getVolume()+transaction.getAmount()<=MAX_VOLUME){
+                    coin.setVolume(coin.getVolume()+transaction.getAmount());
+                    transaction.setTransactionHash(transaction.getBlockHash());
+                }
+            }
+        }
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/services/CryptoService.java:118]: Imagine if BTC is locked for some reason, does ETH coins can be added?

File: backend/assignments/1/src/main/java/org/services/CryptoService.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                coin.setVolume(coin.getVolume()+transaction.getAmount());
+                transaction.setTransactionHash(transaction.getBlockHash());
+            }
+        }
+    }
+
+    /**
+     *
+     * @param symbol
+     * @return
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/services/CryptoService.java:82]: What it returns?
Write proper java docs

File: backend/assignments/1/src/main/java/org/services/CryptoService.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+import org.models.*;
+
+import java.util.*;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+public class CryptoService {
+    private static final int MAX_VOLUME=1000000;
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/services/CryptoService.java:11]: Integer.MAX_VALUE

File: backend/assignments/1/src/main/java/org/services/CryptoApplication.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+        System.out.println("coins.csv File loaded successfully");
+
+        List<Trader> traders=loadTraderFromCSV("src/main/resources/traders.csv");
+        System.out.println("traders.csv File loaded successfully");
+
+        List<Transaction> transactions=loadTransactionsFromJSON("src/main/resources/small_transaction.json");
+        System.out.println("large_transaction.csv File loaded successfully");
+
+        JsonNode jsonNode=loadJsonNodeFromJSON("src/main/resources/small_transaction.json");
+        System.out.println("JsonNode loaded successfully");
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/services/CryptoApplication.java:33]: Could have defined constants and stored these paths

File: backend/assignments/1/src/main/java/org/models/TransactionType.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package org.models;
+
+public enum TransactionType {
+    BUY,
+    SELL,
+    UPDATE_PRICE,
+    ADD_VOLUME
+}
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/models/TransactionType.java:8]: Good!!

File: backend/assignments/1/src/main/java/org/models/Transaction.java [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+        }
+
+        while (transactionHash.length() < 128) {
+            int index = (int) (rnd.nextFloat() * SALTCHARS.length());
+            transactionHash.append(SALTCHARS.charAt(index));
+        }
+
+        String hashCode = transactionHash.toString();
+        return "0x" + hashCode.toLowerCase();
+    }
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/models/Transaction.java:149]: Could have defined this in `utils`. `models` are generally used to define model of data you are storing

File: backend/assignments/1/src/main/java/org/models/Transaction.java [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+    public void setCoinSymbol(String coinSymbol) {
+        this.coinSymbol = coinSymbol;
+    }
+    @Override
+    public boolean equals(Object o){
+        if (this==o)
+            return true;
+        if (o==null||getClass()!=o.getClass())
+            return false;
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/models/Transaction.java:119]: Wrap code blocks in curly braces

File: backend/assignments/1/src/main/java/org/models/Trader.java [hunk 10]
```diff
@@ -0,0 +1,53 @@
+package org.models;
+
+public class Trader {
+    private String id;
+    private String firstName;
+    private String lastName;
+    String phone;
+    private String walletAddress;
+   // private Portfolio portfolio;
```
Comment (Sagun15) [inline backend/assignments/1/src/main/java/org/models/Trader.java:9]: Remove stale variables

**Review by Sagun15 — COMMENTED** [review]
Reach out to me if you have any doubts on comments

**Pull request merged**